shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteCardData.kt

```diff
@@ -259,11 +349,43 @@ data class RouteCardData(private val lineOrRoute: LineOrRoute, val stopData: Lis
         }
 
         fun filterIrrelevantData(
+            filterAtTime: Instant,
             cutoffTime: Instant?,
-            showAllPatternsWhileLoading: Boolean,
-            filterCancellations: Boolean
+            context: Context,
+            globalData: GlobalResponse
         ): ListBuilder {
-            // TODO
+
+            val showAllPatternsWhileLoading = context.isStopDetails()
+            for (entry in this.data) {
+                val (routeOrLineId, byStopId) = entry
+                for (stopEntry in byStopId.stopData) {
+                    val (stopId, byDirectionId) = stopEntry
+                    val isSubway = byStopId.lineOrRoute.isSubway()
+                    for (directionEntry in byDirectionId.data) {}
+
+                    byDirectionId.data =
+                        byDirectionId.data
+                            .filter {
+                                val (directionId, leafBuilder) = it
+                                leafBuilder.shouldShow(
+                                    byDirectionId.stop,
+                                    filterAtTime,
+                                    cutoffTime,
+                                    showAllPatternsWhileLoading,
+                                    isSubway,
+                                    globalData
+                                )
+                            }
+                            .mapValues {
+                                val (directionId, leafBuilder) = it
+                                leafBuilder
+                                    .filterCancellations(isSubway, context)
+                                    .filterArrivalOnly()
```

Comment:
note: `NearbyStaticData.withRealtimeInfo` doesn't appear to filter out arrival only upcoming trips, it would only remove entire headsign entries where all are arrival only. 

It seems worth doing this because the Leaf for a single direction could have a mix of arrival-only trips and trips with departures.